PowerEditor/src/Notepad_plus.cpp

```diff
@@ -1461,11 +1461,11 @@ void Notepad_plus::removeEmptyLine(bool isBlankContained)
 	FindOption env;
 	if (isBlankContained)
 	{
-		env._str2Search = TEXT("^[\\t ]*$(\\r\\n|\\r|\\n)");
+		env._str2Search = TEXT("^([\\t ]*$(\\r?\\n|\\r))(\\1)*");

```

Comment:
Isn't this a better regex here?:  `^([\t ]*\R)+`

Reply:
https://www.boost.org/doc/libs/1_80_0/libs/regex/doc/html/boost_regex/syntax/perl_syntax.html
For `\R`:
>The escape sequence \R matches any line ending character sequence, specifically it is identical to the expression (?>\x0D\x0A?|[\x0A-\x0C\x85\x{2028}\x{2029}]).

It looks like it will work too. I haven't checked exactly which characters represent these, but if necessary this will also be simpler `"^([\\t ]*(\\r?\\n|\\r))+`.

Reply:
Perhaps this, taking a hint from the boost docs you cited:  `^([\x09\x20]*(?>\x0D\x0A?|\x0A))+`